File: challenge-2/challenge-2.rb
inputA = '1c0111001f010100061a024b53535009181c'
inputB = '686974207468652062756c6c277320657965'

bytesA = [inputA].pack('H*').unpack('C*')
bytesB = [inputB].pack('H*').unpack('C*')

output = bytesA.zip(bytesB).map { |a, b|
  a ^ b
}.pack('C*')

puts output.bytes.map { |b| b.to_s(16) }.join

File: challenge-2/challenge-2.go
package main

import (
	"encoding/hex"
	"fmt"
)

func main() {
	inputA, _ := hex.DecodeString("1c0111001f010100061a024b53535009181c")

	inputB, _ := hex.DecodeString("686974207468652062756c6c277320657965")
	var out []byte

	for i := 0; i < len(inputA); i++ {
		out = append(out, inputA[i]^inputB[i])
	}

	fmt.Println(hex.EncodeToString(out))
}
